farmbot_ext/lib/farmbot_ext/amqp/connection_worker.ex

```diff
@@ -4,24 +4,29 @@ defmodule FarmbotExt.AMQP.ConnectionWorker do
   """
 
   use GenServer
-  alias FarmbotCore.Project
-  alias FarmbotExt.JWT
   require Logger
+  alias FarmbotExt.JWT
+  alias FarmbotCore.Project
+
+  @me __MODULE__
+
+  # IMPORTANT- prevents bad builds!:
+  FarmbotExt.fetch_impl!(@me, :network_impl)
```

Comment:
I like this idea, but i think we may want to do it a bit differently. Now there is a possibility of compiler deadlock between modules. [this post](https://elixir-lang.org/blog/2012/04/24/a-peek-inside-elixir-s-parallel-compiler/) will probably give some insight.